SAP2000_Adapter/AdapterActions/Execute.cs

```diff
@@ -121,6 +121,25 @@ public bool RunCommand(ClearResults command)
 
         /***************************************************/
 
+        public bool RunCommand(Exit command)
+        {
+            if (command.SaveBeforeClose)
+            {
+                if (m_app.SapModel.GetModelFilepath() == "(Untitled)")
```

Comment:
```suggestion
                if (m_app.SapModel.GetModelFilename() == "(Untitled)")
```

Reply:
Need just the file name, not full path. I tested with this change and it works properly, failing to close the file and throwing the desired error in Grasshopper. Without it, the error is never hit, and SAP2000 throws up a dialogue about failing to save, then closes after the user accepts.

Reply:
GetModelFilepath() returns "" rather than "(Untitled)", cancelling my comment.